### DISCUSSION TITLE
can connect multiple device to esp32 through bluetooth serial

### DISCUSSION THREAD

**Discussion by hoangtntx** [espressif/arduino-esp32#7979] [General]
i want share data from esp32 to multi device through bluetooth serial but only one device can connect to it. how to modify library to do that. thx :(

**Comment by mrengineer7777** [comment]
Bluetooth was originally a point to point connection.  Newer versions of Bluetooth support multipoint, but compatibility may be an issue.  Either way, I don't think the Arduino layer supports it.

https://espressif-docs.readthedocs-hosted.com/projects/espressif-esp-faq/en/latest/software-framework/ble-bt.html

Have you considered ESP-NOW protocol? https://randomnerdtutorials.com/esp-now-many-to-one-esp32/

**Comment by hoangtntx** [comment]
past year, i modified bluetoothserial library and when 5 phone connect to esp32 and 5 phone able receive data from esp32 through bluetooth serial, but when i update core, bluetoothserial library deleted and i dont remember what i do :(

**Reply by mrengineer7777** [reply]
:'( That's very sad news.  I highly recommend backing up your work.  On some of my projects I just sync them to a private Git repository.  GitHub desktop makes it easy.
